packages/ui/components/scheduler/components/blocking/BlockedList.tsx

```diff
@@ -0,0 +1,111 @@
+import { useId, useMemo } from "react";
+
+import dayjs from "@calcom/dayjs";
+
+import { useSchedulerStore } from "../../state/store";
+import { BlockedTimeCell } from "./BlockedTimeCell";
+
+type Props = {
+  day: dayjs.Dayjs;
+  containerRef: React.RefObject<HTMLDivElement>;
+};
+
+function roundX(x: number, roundBy: number) {
+  return Math.round(x / roundBy) * roundBy;
+}
+
+type BlockedDayProps = {
+  startHour: number;
+  endHour: number;
+  day: dayjs.Dayjs;
+};
+
+function BlockedBeforeToday({ day, startHour, endHour }: BlockedDayProps) {
+  return (
+    <>
+      {day.isBefore(dayjs(), "day") && (
+        <div
+          key={day.format("YYYY-MM-DD")}
+          className="absolute z-40 w-full"
+          style={{
+            top: `var(--one-minute-height)`,
+            zIndex: 60,
+            height: `calc(${(endHour + 1 - startHour) * 60} * var(--one-minute-height))`, // Add 1 to endHour to include the last hour that we add to display the last vertical line
+          }}>
+          <BlockedTimeCell />
+        </div>
+      )}
+    </>
+  );
+}
+function BlockedToday({ day, startHour, gridCellsPerHour }: BlockedDayProps & { gridCellsPerHour: number }) {
+  const dayStart = useMemo(() => day.startOf("day").hour(startHour), [day, startHour]);
+  const nowComparedToDayStart = useMemo(() => dayjs().diff(dayStart, "minutes"), [dayStart]);
+  return (
+    <>
+      {day.isToday() && (
+        <div
+          key={day.format("YYYY-MM-DD")}
+          className="absolute z-40 w-full"
+          style={{
+            top: `var(--one-minute-height)`, // Still need this as this var takes into consideration the offset of the "AllDayEvents" bar
+            zIndex: 60,
+            height: `calc(${roundX(
+              nowComparedToDayStart,
+              60 / gridCellsPerHour
+            )} * var(--one-minute-height) - 2px)`, // We minus the border width to make it 🧹
+          }}>
+          <BlockedTimeCell />
+        </div>
+      )}
+    </>
+  );
+}
```

Comment:
This blocks up to the current time of the day and takes into account grid cells.
e.g. 

- Its currently 12:05 and your grid cells are every 15 minutes
- We block until 12:15 on the grid so you can't be booked outside of your time interval